src/libsyntax/parse/parser.rs

```diff
@@ -7852,116 +7845,6 @@ impl<'a> Parser<'a> {
             Applicability::MaybeIncorrect,
         ).emit();
     }
-
-    /// When lowering a `async fn` to the HIR, we need to move all of the arguments of the function
-    /// into the generated closure so that they are dropped when the future is polled and not when
-    /// it is created.
-    ///
-    /// The arguments of the function are replaced in HIR lowering with the arguments created by
-    /// this function and the statements created here are inserted at the top of the closure body.
-    fn construct_async_arguments(&mut self, asyncness: &mut Spanned<IsAsync>, decl: &mut FnDecl) {
```

Comment:
Great to see this removed.
I somehow missed the moment when all this out-of-place stuff was added to the parser.

Reply:
The replacement in HIR lowering is not done yet, so if you want I can back out the async changes and do that in a future PR.

Reply:
r=me for the part before async then

Reply:
I've submitted #61413 that builds upon this and implements the async part.